runtime/testy/runtime.py

```diff
@@ -47,11 +47,14 @@ def runtime():
       print("Starting studentCode attempt: %s" % (restartCount,))
       spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode)
       while True:
-        newBadThing = badThingsQueue.get(block=True)
-        print(RUNTIME_CONFIG.DEBUG_DELIMITER_STRING.value)
-        print(newBadThing)
-        if newBadThing.event in restartEvents:
-          break
+        try:
```

Comment:
Can you explain why this hunk was changed?

It looks like these two pieces of code should behave identically, with the old code being simpler. If that's not the case that means there's something interesting going on, that deserves careful thought.